src/main/java/com/example/Restaurant_Management_System_REST_API/model/entity/Customer.java

```diff
@@ -0,0 +1,73 @@
+package com.example.Restaurant_Management_System_REST_API.model.entity;
+
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+@Entity
+@Table(name = "application_users")
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class Customer implements UserDetails {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", nullable = false)
+    private Long id;
+
+    //Below are fields for security issues
+    private String password;
+    private String emailAddress; //TODO: delete when abstract class with contact details will be implemented
+    @ElementCollection(fetch = FetchType.EAGER)
+    private Set<String> roles;
```

Comment:
What's the purpose of this field with that type? Could you explain to me why is it made here?

Reply:
I want to keep somewhere those roles. Why? what's the case?